Classes/Bookmark v2/BookmarkViewController.swift

```diff
@@ -0,0 +1,304 @@
+//
+//  BookmarkViewController.swift
+//  Freetime
+//
+//  Created by Hesham Salman on 11/5/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+import IGListKit
+import Apollo
+
+class BookmarkViewController: UIViewController,
+    ListAdapterDataSource,
+    PrimaryViewController,
+    UISearchBarDelegate,
+BookmarkHeaderSectionControllerDelegate,
+BookmarkListener,
+BookmarkSectionControllerDelegate,
+InitialEmptyViewDelegate,
+TabNavRootViewControllerType {
+
+    private let client: GithubClient
+    private let headerKey = "com.freetime.BookmarkViewController.bookmark-header-key" as ListDiffable
+
+    private let searchBar = UISearchBar()
+    private lazy var adapter: ListAdapter = { ListAdapter(updater: ListAdapterUpdater(), viewController: self) }()
+    private let collectionView: UICollectionView = {
+        let view = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+        view.alwaysBounceVertical = true
+        view.backgroundColor = Styles.Colors.background
+        return view
+    }()
+    private var originalContentInset: UIEdgeInsets = .zero
+
+    var bookmarkStore: BookmarkStore
+
+    enum State {
+        case idle
+        case filtering(String)
+    }
+
+    var state: State = .idle
+
+    // MARK: - Initialization
+
+    init(client: GithubClient) {
+        self.client = client
+        guard let store = client.bookmarksStore else { fatalError("Client does not have a bookmark store") }
```

Comment:
What happens on fresh app install? Do we not init the VC? Maybe the init should be failable so that is represented?